src/main/asciidoc/driver/configuration.adoc

```diff
@@ -189,3 +195,54 @@ include::{python-examples}/test_examples.py[tag=tls-signed]
 ----
 --
 ====
+
+
+=== Authenticate user
+
+While the encrypted communication is established between a Neo4j driver and the server, the server may also require the driver to provide authentication credentials for the user connecting to the database.
+Authentication credentials are required if the server is configured to have authentication enabled.
+// See 'Securing Neo4j Server' (http://neo4j.com/docs/3.0.0-SNAPSHOT/security-server.html) for more information.
+
+[.tabbed-example]
+.Example: Connecting the driver to the server with authentication credentials
+====
+[include-with-dotnet]
+--
+[source, csharp]
+----
+include::{dotnet-examples}/Examples.cs[tags=connect-with-basic-auth]
+----
+--
+
+[include-with-java]
+--
+[source, java]
+----
+include::{java-examples}/Examples.java[tags=connect-with-basic-auth]
+----
+--
+[include-with-javascript]
+--
+[source, javascript]
+----
+include::{javascript-examples}/examples.test.js[tags=connect-with-basic-auth]
+----
+--
+[include-with-python]
+--
+[source, python]
+----
+include::{python-examples}/test_examples.py[tag=connect-with-basic-auth]
+----
+--
+====
+
+If communication between driver and server are is not encrypted, authentication credentials are sent as plain text.
+It is highly recommended to use authentication with encryption.
+
+[NOTE]
+--
+When connecting to Neo4j for the first time, the user must change the default password.
+Given a fresh installed Neo4j server, if a driver tries to connect to the Neo4j server with the default password, the attempt would probably fail with an error saying that the credentials are expired and a password reset is required.
```

Comment:
I'm not very comfortable with "this will probably fail"! We should make sure we know what the user will experience when using a driver on a new installation of Neo4j, and if they need to take any extra steps to change the password before they can do it, we should really make that super clear.